src/category/category.ts

```diff
@@ -15,13 +15,11 @@ export interface CategoryProps {
   itemExtraDefs?: {
     [name: string]: CategoryItemExtraDef
   }
-  name: string
+  name?: string
 }
 
 export interface Category extends ArenaObject<CategoryProps> {
   id?: number
-  levels?: {
-    [key: number]: CategoryLevel
-  }
+  levels?: CategoryLevel[]
```

Comment:
levels is an indexed object as before, not an array

Reply:
// ====== CREATE
export const newCategory = (props = {}, levels = null) => {
  const category = {
    [keys.uuid]: uuidv4(),
    props,
  }
  return {
    ...category,
    [keys.levels]: levels || [newLevel(category)],
  }
}

Reply:
```
export const newLevel = (category, props = {}, index = R.pipe(getLevels, R.keys, R.length)(category)) => ({
  [CategoryLevel.keys.uuid]: uuidv4(),
  [CategoryLevel.keys.categoryUuid]: ObjectUtils.getUuid(category),
  [CategoryLevel.keys.index]: index,
  [CategoryLevel.keys.props]: {
    [CategoryLevel.keysProps.name]: `level_${index + 1}`,
    ...props,
  },
})```